lattice_controller_engine/lib/lattice_controller_engine/reconciler/spreadscaler.ex

```diff
@@ -0,0 +1,126 @@
+defmodule LatticeControllerEngine.Reconciler.SpreadScaler do
+  alias LatticeControllerEngine.Reconciler.Command
+  alias LatticeControllerEngine.Model.{SpreadScaler, WeightedTarget}
+  alias LatticeControllerEngine.Observed.Lattice
+  alias LatticeControllerEngine.Reconciler.AppSpec, as: AppSpecReconciler
+
+  @type target_instance_map :: %{required(String.t()) => [Instance.t()]}
+
+  @spec perform_spread(
+          String.t(),
+          String.t(),
+          String.t() | nil,
+          SpreadScaler.t(),
+          Lattice.t(),
+          integer()
+        ) :: [Command.t()]
+  def perform_spread(
+        spec_name,
+        image,
+        component_pk,
+        scaler = %SpreadScaler{},
+        actual = %Lattice{},
+        max_per_host \\ 1000,
+        opts \\ %{}
+      ) do
+    for wtarget <- scaler.spread do
+      candidate_hosts = AppSpecReconciler.matching_hosts(actual, wtarget.requirements)
+
+      running_instances =
+        Lattice.running_instances(actual, component_pk, spec_name)
+        |> Enum.filter(fn %{host_id: hid} -> hid in candidate_hosts end)
+
+      desired_instances = trunc(wtarget.weight / 100 * scaler.replicas)
```

Comment:
PEMDAS scares me here, does elixir evaluate this left to right?

Reply:
`*` and `/` have the same precedence so this is left-to-right.  It's doing the percentage (/100) first, e.g. 50 / 100 * 6 == 0.5 * 6 == 3, and not 50 / 600 == 0.083333

Reply:
I had parens there but `mix format` actually removes them.

Reply:
Cool, just making sure. This is one of those bugs that could take hours and would make you want to smash your machine, so I figured I'd just double check